composeApp/src/desktopMain/kotlin/org/ooni/probe/Main.kt

```diff
@@ -21,16 +21,27 @@ import ooniprobe.composeapp.generated.resources.app_name
 import ooniprobe.composeapp.generated.resources.tray_icon
 import org.jetbrains.compose.resources.painterResource
 import org.jetbrains.compose.resources.stringResource
+import org.ooni.probe.data.DeepLinkHandler
 import org.ooni.probe.data.models.DeepLink
+import org.ooni.probe.shared.DesktopOS
+import org.ooni.probe.shared.Platform
 import java.awt.Desktop
 
-fun main() {
-    val autoLaunch = AutoLaunch(appPackageName = "org.openobservatory.ooniprobe")
+val APP_ID = "org.openobservatory.ooniprobe"
+
+fun main(args: Array<String>) {
+    val autoLaunch = AutoLaunch(appPackageName = APP_ID)
 
     val deepLinkFlow = MutableSharedFlow<DeepLink?>(extraBufferCapacity = 1)
 
-    Desktop.getDesktop().setOpenURIHandler { event ->
-        deepLinkFlow.tryEmit(DeepLink.AddDescriptor(event.uri.path.split("/").last()))
+    // Initialize the deep link handler
+    val deepLinkHandler = DeepLinkHandler()
+    deepLinkHandler.initialize(args)
+
+    if ((dependencies.platformInfo.platform as? Platform.Desktop)?.os == DesktopOS.Mac) {
```

Comment:
Do we also need `registerLinuxProtocolHandler()` for mac? Because right now it looks to be running too no?